src/DocChecks.jl

```diff
@@ -405,4 +408,38 @@ else
     footnotes(doc::Documents.Document) = nothing
 end
 
+
+# Link Checks.
+# ------------
+
+import Requests
+
+function linkcheck(doc::Documents.Document)
+    if doc.user.linkcheck
+        println(" > checking external URLs:")
+        for (src, page) in doc.internal.pages
+            println("   - ", src)
+            for element in page.elements
+                Walkers.walk(page.globals.meta, page.mapping[element]) do block
+                    linkcheck(block, doc)
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+function linkcheck(link::Base.Markdown.Link, doc::Documents.Document)
+    if !(link in doc.internal.locallinks)
+        local good = try
+            Requests.statuscode(Requests.head(link.url)) == 200
+        catch err
+            false
+        end
+        print_with_color(good ? :green : :red, good ? "     " : "   X ", link.url, "\n")
```

Comment:
I'll add an extra char.

> Also, I am wondering if, for consistency, we should also show the 200s.

The intention was to make it inconsistent so that it's easier to tell (without colour) whether the link was good.

> why the link was considered a success

If is succeeds then they shouldn't really need to be concerned about why (I'd assume), only when it fails.